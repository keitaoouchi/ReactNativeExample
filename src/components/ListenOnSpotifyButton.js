import React from "react";
import {Linking, Text, TouchableOpacity, View} from "react-native";
import Color from "../misc/Color";

export default ListenOnSpotifyButton = ({url, color, title = 'Spotifyで聴く'}) => (
  <TouchableOpacity
    onPress={() => {
      if (!!url && Linking.canOpenURL(url)) {
        Linking.openURL(url);
      }
    }}
  >
    <View style={ [style.button, !!color && {backgroundColor: color}] }>
      <Text style={ style.text }>{title}</Text>
    </View>
  </TouchableOpacity>
);

const style = {
  button: {
    backgroundColor: Color.green,
    height: 44,
    width: 200,
    padding: 8,
    borderRadius: 22,
    margin: 12,
  },
  text: {
    color: 'white',
    textAlign: 'center',
    fontWeight: 'bold',
    fontSize: 16,
    margin: 4,
  },
};