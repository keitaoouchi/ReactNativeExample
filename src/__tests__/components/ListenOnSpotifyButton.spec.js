import "react-native";
import React from "react";
import renderer from "react-test-renderer";
import ListenOnSpotifyButton from "../../../src/components/ListenOnSpotifyButton";

describe('ListenOnSpotifyButton', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <ListenOnSpotifyButton url="https://example.com"/>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
