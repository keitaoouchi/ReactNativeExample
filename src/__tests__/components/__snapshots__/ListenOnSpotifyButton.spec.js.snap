// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListenOnSpotifyButton renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#1ed760",
          "borderRadius": 22,
          "height": 44,
          "margin": 12,
          "padding": 8,
          "width": 200,
        },
        false,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
          "fontSize": 16,
          "fontWeight": "bold",
          "margin": 4,
          "textAlign": "center",
        }
      }
    >
      Spotifyで聴く
    </Text>
  </View>
</View>
`;
